LPD-XXX

victorg1991 opened the pull request.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

sandrodw3 commented:
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/84690. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

shuyangzhou committed:
LPD-64707 Add support for in-memory filtering

shuyangzhou committed:
LPD-64707 Add in-memory filter support in generated persistence impl.

wanderlast committed:
LPD-61751 add finder

wanderlast committed:
LPD-61751 buildService

wanderlast committed:
LPD-61751 semver

victorKammerer committed:
LPD-62185 Email notification must not be send without notification recipients

dantewang committed:
LPS-63790 Inline HttpServletRequestWrapperImpl and HttpServletResponseWrapperImpl and make SF happy

BarbaraCabrera committed:
LPD-61879 Replace namespace with a token to update the value with the new namespace once it is added or imported

oliv-yu committed:
LPD-58618 site-cms-site-initializer: Update empty state description

oliv-yu committed:
LPD-58618 portal-language-lang: Update wording for empty state description

shuyangzhou committed:
LPD-64707 Regeneration

BarbaraCabrera committed:
LPD-61879 Add necessary integration test to check that the namespace is replaced with the token

BarbaraCabrera committed:
LPD-61879 Add necessary integration test to check that the token is replaced with the namespace

shuyangzhou committed:
LPD-64707 Add service method to expose findByC_N_S()

shuyangzhou committed:
LPD-64707 Regeneration

oliv-yu committed:
LPD-58618 playwright: Include test for drag-drop file in All section

oliv-yu committed:
LPD-58618 portal-language-lang: Remove unused key now

oliv-yu committed:
LPD-58618 portal-language-lang: gw buildLang

RoselaineMarquesMontero committed:
LPD-62162 Make LayoutPageTemplateCollection indexable

brookedalton committed:
LPD-59803 Add tests

brookedalton committed:
LPD-59803 Implement getOrAddEmptyLayout

brookedalton committed:
LPD-59803 ant build-service-portal

brookedalton committed:
LPD-59803 Add empty status if empty model

AliciaGarciaGarcia committed:
LPD-64010 when the object definition defines the permission for the CMS_ADMIN, add the permission for the GUEST and USER roles.

oliv-yu committed:
LPD-58618 site-cms-site-initializer: Disable drag/drop for content folder pages

dantewang committed:
LPS-63790 Apply

dantewang committed:
LPS-63790 SF, rename

dantewang committed:
LPS-63790 SF, explicitly use LiferayContextController and LiferayDispatchTargets

dantewang committed:
LPS-63790 SF, this internal class and related methods are not used anywhere

dantewang committed:
LPS-63790 SF, rename

dantewang committed:
LPS-63790 SF, avoid duplicate logic

dantewang committed:
LPS-63790 SF, use constants where possible

Mikellorza committed:
LPD-58618 Fix failing test deleting unnecessary method

AliciaGarciaGarcia committed:
LPD-64010 add test to check that the permissions are set when a system object definition is published

oliv-yu committed:
LPD-58618 site-cms-site-initializer: SF after rebase

abesousa committed:
LPD-63539 buildService

abesousa committed:
LPD-63539 Add method to link Object Definition and Workflow Definition

abesousa committed:
LPD-63539 Fix compile errors

dantewang committed:
LPS-63790 Fix ClassCastException due to wrong SF, the method may return a ResourceRegistration which is also a EndpointRegistration.

shuyangzhou committed:
LPD-64808 Remove useless companyId swapping, PropsUtil.get() is not company aware.

shuyangzhou committed:
LPD-64808 Remove the aweful PortalRunMode.isTestMode() testing branch. Sacrificing production code for testing support is showing weakness of design abilities. A good test strategy does not intrude production code.

MarinhoFeliphe committed:
LPD-60127 refactor: show portlet of object definitions that are root descendat because they can be created as standalone

MarinhoFeliphe committed:
LPD-60127 test: ensure getBackURL retrieves a different URL depending on the level of the node

MarinhoFeliphe committed:
LPD-60127 fix: an object definition is not enough to define if the entry is descendant or not, check based on the entry itself

MarinhoFeliphe committed:
LPD-60127 fix: model resource permission is already aware of the new root model permission rule based on the root object entry, so there is no need to check if the object definition is descendant, portlet resource permission cannot find out the root object entry id because permissions checked in there are perfomed before the object entry is created

MarinhoFeliphe committed:
LPD-60127 feat: do not include in the layout edge relationship field that are not set because they do not belongs to the context

MarinhoFeliphe committed:
LPD-60127 feat: the API URL to perform CRUD operations in a context that an object definition is a root descendant node varies, pass objectRelationshipID and parentObjectEntryERC in order to adapt getAPIURL and getBackURL

MarinhoFeliphe committed:
LPD-60127 fix: obtain the parent object entry by using service instead of the method of the same layer because the parent can be related or not, so to avoid aditional logic to know if the parent is or not a descendant obtain it via service

MarinhoFeliphe committed:
LPD-60127 refactor: permissions of descendant nodes should not be hidden or nested, they should appear the same way regular object definitions because it can be used for standalone object entries

# breaking

## What modules/apps/object/object-api/src/main/java/com/liferay/object/definition/security/permission/resource/util/ObjectDefinitionResourcePermissionUtil.java

Removing unnecessary parameters from many methods

## Why

Logic is no longer applied

----

RoselaineMarquesMontero committed:
LPD-62162 Add IndexerReindexTest

victorg1991 committed:
LPD-62003 Allow setting inputReadOnly in the configuration of the inputs

victorg1991 committed:
LPD-62003 Select if the inputs should be read only

# breaking

## What modules/apps/layout/layout-api/src/main/java/com/liferay/layout/manager/FormManager.java

Allow rendering the inputs as read only when filling the form

## Why

For side by side translation view we need one of the sides to be represented as a read only form

----

victorg1991 committed:
LPD-62003 Adapt code

victorg1991 committed:
LPD-62003 Rename

victorg1991 committed:
LPD-62003 Add master layout for translation

MarinhoFeliphe committed:
LPD-60127 test: ensure getAPIURL takes root model context into account

AliciaGarciaGarcia committed:
LPD-64010 now the definitions has permissions so you can see 4 options

victorg1991 committed:
LPD-62003 Add Struts action for translating that will create display page if not created

victorg1991 committed:
LPD-62003 Add option to translate content

victorg1991 committed:
LPD-62003 Extract constant

victorg1991 committed:
LPD-62003 Add reset translation display page struts action

victorg1991 committed:
LPD-62003 Reset translation page after publish

abesousa committed:
LPD-63539 Add WorkflowDefinitionLinkUtil

abesousa committed:
LPD-63539 Add parameter to link Object Definition and Workflow Definition

# breaking

## What modules/apps/object/object-api/src/main/java/com/liferay/object/service/ObjectDefinitionService.java

The methods addCustomObjectDefinition, addSystemObjectDefinition, updateCustomObjectDefinition and updateSystemObjectDefinition from ObjectDefinitionService now require an additional parameter List workflowDefinitionLinks

## Why

We need this parameter to be able to link Workflow Definitions to the Object Definition being added or updated

----

AliciaGarciaGarcia committed:
LPD-64010 do not iterate on the array

brookedalton committed:
LPD-59803 Add method to remote API and verify permissions

brookedalton committed:
LPD-59803 ant build-service-portal

CaioERP committed:
LPD-64024 bring back the statuses and create Get method

shuyangzhou committed:
LPD-64808 Inline _withFeatureFlagManager()

shuyangzhou committed:
LPD-64808 Inline FeatureFlagsBagProvider.withFeatureFlagsBag()

shuyangzhou committed:
LPD-64808 Mimic the FeatureFlagManagerUtil's original PortalRunMode.isTestMode() logic (directly reading portal.properties for feature flag settings, ignoring PreferenceAwareFeatureFlag and DependencyAwareFeatureFlag). This original logic is technically wrong, but there are way too many integration tests relying on this bug to work, keep this same behavior in FeatureFlagTestRule to avoid massive refactor for integration tests

wanderlast committed:
LPD-61751 add service layer call

wanderlast committed:
LPD-61751 build service

wanderlast committed:
LPD-61751 use new method

achaparro committed:
LPD-59803 We can not expose the userId in the remote service layer

achaparro committed:
LPD-59803 Build service

achaparro committed:
LPD-59803 GetOrAddIncomplete layout when importing a page with non-existent parent

achaparro committed:
LPD-59803 Support incompatible status for layout search index

wanderlast committed:
LPD-61751 SF

victorg1991 committed:
LPD-62003 Add Test

AliciaGarciaGarcia committed:
LPD-64415 reset Company Locales

Mikellorza committed:
LPD-58618 Fix test

victorg1991 committed:
LPD-62003 Update test

victorg1991 committed:
LPD-62003 Fix call

victorg1991 committed:
LPD-62003 Baseline

rodrigoguedes committed:
LPD-56346 headless-admin-taxonomy-impl: implement ExportImportVulcanBatchEngineTaskItemDelegate
https://liferay.atlassian.net/browse/LPD-56346

rodrigoguedes committed:
LPD-56346 asset-categories-test: create test
https://liferay.atlassian.net/browse/LPD-56346

wanderlast committed:
LPD-61751 baseline

robertoDiaz committed:
LPD-64510 Fix typo

robertoDiaz committed:
LPD-64510 Add missing case

robertoDiaz committed:
LPD-64510 Simplify

robertoDiaz committed:
LPD-64510 Reorder

achaparro committed:
LPD-59803 Simplify and check first the one more restricted

MarinhoFeliphe committed:
LPD-60127 feat: do not display relationship object fields that belongs to an edge relationship

thektan committed:
LPD-64585 playwright: Fix count to 14 since an additional item was added

kevenleone committed:
LPD-58558 Update Fragment configuration

victorg1991 committed:
LPD-62003 Add CSS class to the main form to differentiate it

victorg1991 committed:
LPD-62003 Don't create constant file

shuyangzhou committed:
LPD-64811 Optimize GroupImpl's classname type detection

shuyangzhou committed:
LPD-64811 Regenerate

victorKammerer committed:
LPD-62185 Unit Test to check if notification recipient map is empty if no recipient is provided

abesousa committed:
LPD-63539 Add tests

MarinhoFeliphe committed:
LPD-60127 style: associate the return with the case handled by the condition

MarinhoFeliphe committed:
LPD-60127 style: rename variables

thektan committed:
LPD-64585 playwright: Title is now an action link

Mikellorza committed:
LPD-58618 site-cms-site-initializer: Apply drag and drop into propsTransformer for all/files/folder sections

Mikellorza committed:
LPD-58618 Update test

jkappler committed:
LPD-64543 Adds new language key

jkappler committed:
LPD-64543 Build lang

jkappler committed:
LPD-64543 Use new language key

Mikellorza committed:
LPD-64544 Change delete action in order to show a custom confirm modal instead of the default one from the FDS

Mikellorza committed:
LPD-64544 Add language keys

Mikellorza committed:
LPD-64544 Add delete confirmation modal

Mikellorza committed:
LPD-64544 buildLang

MarinhoFeliphe committed:
LPD-60127 generated: baseline

CaioERP committed:
LPD-64024 Sort

thiago-buarqque committed:
LPD-63217 folder filter should be made based on the object definition, not on the object entry folder

ealonso committed:
LPD-64757 Adds test to expose the issue

ealonso committed:
LPD-64757 We need to call super method to check permissions

thiago-buarqque committed:
LPD-63217 only run batch task for non-standard tests, that is the default ones in base classes. Batch task takes 1 to 2 seconds to run

abfranca committed:
LRSD-9587 Add new roles to CP for article management

thiago-buarqque committed:
LPD-63217 remove duplicated feature flag and take advantage of annotation's default parameter

thiago-buarqque committed:
LPD-63217 handle postgresql and hypersonic database when querying localized title

thiago-buarqque committed:
LPD-63217 centralize logic that returns the expressions for extracting title and localized title for ObjectEntryVersion

shuyangzhou committed:
LPD-64812 Avoid parsing plain text as html

shuyangzhou committed:
LPD-64813 Cache jsoup document

abesousa committed:
LPD-63539 SF

kevenleone committed:
LPD-64772 Fix of filter in date range in admin dashboard (orders page) that was not counting the end date range

kevenleone committed:
LPD-64772 Fix percentage in admin dashboard in donut charts higher than 100%

kevenleone committed:
LPD-64772 wrong yearly count in admin dashboard

kevenleone committed:
LPD-64772 In the SSA Dashboard the “New Potential Expiration Date” is not adding the new duration days

kevenleone committed:
LPD-64772 Minor adjust to Marketplace fragments and templates

kevenleone committed:
LPD-64772 SF

MarinhoFeliphe committed:
LPD-60127 fix: if the object entry is a root descendant node the model name have to be the same as its root

felipelorenz committed:
LPD-64768 Revert "LPD-62960 SF"

This reverts commit 83a2860c4ab7019c66d363c7aced6a90c9def132.

achaparro committed:
LPD-62162 Not needed

timscottbell committed:
LPD-64782 Add clusterworkflowtemplate

achaparro committed:
LPD-62162 Use existing group due to performance reasons

achaparro committed:
LPD-62162 Test the update case too

achaparro committed:
LPD-62162 SF

MarinhoFeliphe committed:
LPD-60127 test: fix tests

shuyangzhou committed:
LPD-64808 Swap test rule wrapping order. FeatureFlagTestRule now uses FeatureFlagManagerUtil which depends on JSONFactoryUtil

MarinhoFeliphe committed:
LPD-60127 feat: standalone object actions can also be added to a definition that is not a root because those definitions can have standalone object entries

MarinhoFeliphe committed:
LPD-60127 fix: move the logic to obtain the HTTP method to display context and ensure that if an entry is a root descendant node a PATCH will be perform

achaparro committed:
LPD-59803 Content pages have also parent layout

achaparro committed:
LPD-59803 Apply usages. Regular content pages have parent, templates use the default parent id

achaparro committed:
LPD-59803 Baseline

achaparro committed:
LPD-59803 Since this method can not be call if LazyReferencingThreadLocal is not enabled, we can set layout.instanceable.allowed to true here

achaparro committed:
LPD-59803 Not needed

brianchandotcom committed:
LPD-64707 Rename

brianchandotcom committed:
LPD-64707 Rename

brianchandotcom committed:
LPD-62003 SF

achaparro committed:
LPD-59803 Right order

achaparro committed:
LPD-59803 Apply

achaparro committed:
LPD-62162 Remove Fixture

achaparro committed:
LPD-62162 Test also deleting using the service. It improves performance since DataGuard does nothing

achaparro committed:
LPD-62162 Inline

brianchandotcom committed:
LPD-62162 Rename

brianchandotcom committed:
LPD-62162 Rename

brianchandotcom committed:
LPD-62162 Rename

brianchandotcom committed:
LPD-62162 auto SF

ling-alan-huang committed:
LPD-63871 Auto SF

shuyangzhou committed:
LPD-64808 Simplify tests

shuyangzhou committed:
LPD-64707 Partial revert bffdeeaf18618a561933b4df59e2eff8898d5316, holding signedIn and userId as local variables is on purpose. At this point the permissionChecker object has many layers of wrappers, repeatedly calling these 2 getters with deep call stacks when the freach loop is large can be very costly.

shuyangzhou committed:
LPD-64707 Rename var/parameter for consistency.

kevenleone committed:
LPD-64772 Simplify mutate call

shuyangzhou committed:
LPD-64707 Semantic versioning

MarinhoFeliphe committed:
LPD-60127 test: fix testGetObjectEntryByExternalReferenceCodeWithSlash

MarinhoFeliphe committed:
LPD-60127 text: remove coverage of deprecated logic

kevenleone committed:
LPD-64772 Update DashboardNavigation size

kevenleone committed:
LPD-64772 Rename

achaparro committed:
LPD-59803 Redundant. AddLayout already reindex.

achaparro committed:
LPD-59803 Build service

liferay-translation committed:
LPS-77699 Update Translations

thiago-buarqque committed:
LPD-63217 baseline

dantewang committed:
LPS-63790 SF, buy space

ruben-pulido committed:
LPD-64831 baseline

jkappler committed:
LPD-64738 Cover image for blogs shouldn't be required

sandrodw3 committed:
LPD-62003 Enable localization logic even if it's read only

brianchandotcom committed:
LPD-64813 SF

bakayattila committed:
Revert "LPD-63819 Only display Item selector with group permissions"

This reverts commit e145371171c2d0ee03d8745e69de877a9d2856ad.

smottal committed:
LPD-64886 Fix locators

brianchandotcom committed:
LPD-64757 Together

brianchandotcom committed:
LPD-64811 Do we have to do it like this?

brianchandotcom committed:
LPD-64544 Wordsmith

brianchandotcom committed:
LPD-64544 Regen

thiago-buarqque committed:
LPD-63217 SF

brianchandotcom committed:
LPD-60127 SF

brianchandotcom committed:
LPD-64772 SF

brianchandotcom committed:
LPD-61879 SF

brianchandotcom committed:
LPD-61879 Can I change it to this?

brianchandotcom committed:
LPD-61879 SF

brianchandotcom committed:
LPD-64782 Sublime case sensitive sort

RoselaineMarquesMontero committed:
LPD-62162 Add REINDEX to interface LayoutPageTemplateCollectionLocalService

brianchandotcom committed:
LPD-62185 As used

brianchandotcom committed:
LPD-64010 Rename

brianchandotcom committed:
LPD-64010 Sort

brianchandotcom committed:
LPD-63539 SF

brianchandotcom committed:
LPD-63539 Sort

brianchandotcom committed:
LPD-56346 SF

brianchandotcom committed:
LPD-64707 As used

brianchandotcom committed:
LPD-64707 SF

brianchandotcom committed:
LPD-64707 prep next

brianchandotcom committed:
LPD-64190 prep next

brianchandotcom committed:
LPD-64707 LPD-64190 prep next

victorg1991 committed:
LPD-64979 Add method to add one specific fragment to the form

victorg1991 committed:
LPD-64979 Include the inline input for the title and add correct spacing between elements

victorg1991 committed:
LPD-64979 Update translation layout

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.
